helm/cluster-autoscaler-app/templates/deployment.yaml

```diff
@@ -43,6 +43,8 @@ spec:
             - --cloud-provider={{ .Values.provider }}
             - --expander={{ .Values.configmap.expander }}
             - --logtostderr=true
+            # Don't use auto discovery in integration tests.
+            {{- if eq .Values.e2e false }}
```

Comment:
The test is fine with using `aws` as the provider. But if auto discovery is on then the pod crashes and the test fails. 

This is the best way I can see to prevent it.